src/masternodes/mn_checks.cpp

```diff
@@ -3072,22 +3072,27 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor
         if (!collaterals)
             return Res::Err("Vault with id %s has no collaterals", obj.vaultId.GetHex());
 
+        auto attributes = mnview.GetAttributes();
+        assert(attributes);
+        CDataStructureV0 loanKey{AttributeTypes::Live, ParamIDs::Economy, EconomyKeys::Loans};
+        auto balances = attributes->GetValue(loanKey, CBalances{});
```

Comment:
for easier reading: could we rename that to "loanAttributeBalances"?

the `balances.add(loanAmount)` looks like there is something added to the account balance etc. Feels like a future-refactor-error in the making.